net-libs/iojs/iojs-1.6.4.ebuild

```diff
@@ -0,0 +1,129 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Header: $
+
+EAPI=5
+
+PYTHON_COMPAT=( python2_7 )
+
+MY_PV="v${PV}"
+MY_P="${PN}-${MY_PV}"
+
+inherit flag-o-matic pax-utils python-any-r1 toolchain-funcs
+
+DESCRIPTION="An npm compatible platform originally based on node.js"
+HOMEPAGE="http://iojs.org/"
+SRC_URI="http://iojs.org/dist/${MY_PV}/${MY_P}.tar.xz"
+
+LICENSE="Apache-1.1 Apache-2.0 BSD BSD-2 MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~x86 ~x64-macos"
+IUSE="bundled-libs debug icu +npm snapshot +ssl"
+
+RDEPEND="icu? ( dev-libs/icu )
+	!bundled-libs? (
+		>=net-libs/http-parser-2.3
+		>=dev-libs/libuv-1.4.2
+		>=dev-libs/openssl-1.0.1m[-bindist]
+	)"
+DEPEND="${RDEPEND}
+	${PYTHON_DEPS}
+	!!net-libs/nodejs"
+S="${WORKDIR}/${MY_P}"
+
+pkg_pretend() {
+	if ! test-flag-CXX -std=c++11 ; then
+		die "Your compiler doesn't support C++11. Use GCC 4.8, Clang 3.3 or newer."
+	fi
+}
+
+src_prepare() {
+	tc-export CC CXX PKG_CONFIG
+	export V=1 # Verbose build
+	export BUILDTYPE=Release
+
+	# fix compilation on Darwin
+	# http://code.google.com/p/gyp/issues/detail?id=260
+	sed -i -e "/append('-arch/d" tools/gyp/pylib/gyp/xcode_emulation.py || die
+
+	# make sure we use python2.* while using gyp
+	sed -i -e "s/python/python2/" deps/npm/node_modules/node-gyp/gyp/gyp || die
+	sed -i -e "s/|| 'python'/|| 'python2'/" deps/npm/node_modules/node-gyp/lib/configure.js || die
+
+	# less verbose install output (stating the same as portage, basically)
+	sed -i -e "/print/d" tools/install.py || die
+
+	# proper libdir, hat tip @ryanpcmcquen https://github.com/iojs/io.js/issues/504
+	local LIBDIR=$(get_libdir)
+	sed -i -e "s|lib/|${LIBDIR}/|g" tools/install.py || die
+	sed -i -e "s/'lib'/'${LIBDIR}'/" lib/module.js || die
+	sed -i -e "s|\"lib\"|\"${LIBDIR}\"|" deps/npm/lib/npm.js || die
+
+	epatch "${FILESDIR}"/${PN}-1.2.0-pkgconfig.patch
+
+	# Avoid a test that I've only been able to reproduce from emerge. It doesnt
+	# seem sandbox related either (invoking it from a sandbox works fine).
+	# The issue is that no stdin handle is openened when asked for one.
+	# It doesn't really belong upstream , so it'll just be removed until someone
+	# with more gentoo-knowledge than me (jbergstroem) figures it out.
+	rm test/parallel/test-stdout-close-unref.js
+
+	# debug builds. change install path, remove optimisations and override buildtype
+	if use debug; then
+		sed -i -e "s|out/Release/|out/Debug/|g" tools/install.py
+		filter-flags -O*
```

Comment:
Don't do that. `CFLAGS` are meant to be independent of `USE=debug`. One needs to explicitly specify `-O -g` options he needs.